File: PTA/Basic_Level/1004.cpp
#include<bits/stdc++.h>
using namespace std;


int main() {
	int max = -1, min = 101, score;
	int n;
	string num, name, max_num, max_name, min_num, min_name;
		cin >> n;
	for(int i = 0; i < n; ++i) {
		cin >> name >> num >> score;
		if (score > max) {
			max_name = name;
			max_num = num;
			max = score;
		} 
		if (score < min) {
			min_name = name;
			min_num = num;
			min = score;
		}
	}
	cout << max_name << " " << max_num << endl;
	cout << min_name << " " << min_num << endl;
	return 0;
	
}

File: PTA/Basic_Level/1004.rs
use std::io::stdin;
fn main() {
    let mut input: String = String::new();
    let mut maxscore: i32 = -1;
    let mut minscore: i32 = 101;
    let mut maxnum: String = String::new();
    let mut minnum: String = String::new();
    let mut maxname: String = String::new();
    let mut minname: String = String::new();

    stdin().read_line(&mut input).unwrap();
    let mut n: i32 = input.trim().parse().unwrap();
    input = "".to_string();
    for i in 0..n {
        stdin().read_line(&mut input).unwrap();
        //println!("{} i: {}", input, i);
        let v: Vec<String> = input.trim().split(' ').map(|s| s.to_string()).collect();
        let score: i32 = v[2].parse().unwrap();
        if score > maxscore {
            maxnum = v[1].clone();
            maxname = v[0].clone();
            maxscore = score;
        }
        if score < minscore {
            minnum = v[1].clone();
            minname = v[0].clone();
            minscore = score;
        }
        input = "".to_string();
    }
    println!("{} {}", maxname, maxnum);
    println!("{} {}", minname, minnum);
}
